---
title: "Analysis ITS mutants part 2/2"
author: "Harriet"
date: "24/03/2021"
output:
  word_document: default
  html_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE) #prints code
```

This is part 2/2 of the code for ITS amplicon analyses. See part 1/2 for general compositional data.

First, all the necessary packages were loaded and appropriate files called.

```{r}
library(phyloseq)
library(data.table)
library(vegan)
library(ggplot2)
library(readxl)
library(dplyr)

biom_file<-import_biom("~/metabarcoding_miPEP_mutants/ITS/ITS/manip_mutants/export/otu_tables/otu_table_filtered.biom","~/metabarcoding_miPEP_mutants/ITS/ITS/manip_mutants/export/tree/tree.fasttree","~/metabarcoding_miPEP_mutants/ITS/ITS/manip_mutants/export/otu_tables/otus.fasta", parseFunction=parse_taxonomy_greengenes)
biom_file
```

A data frame was created from the mapping_file, which contains all the information concerning each sample. Each row corresponds to a sample and was named in consequence. Merge the biom file and the metadata into one phyloseq object.

```{r}
sample<- data.frame(fread("~/metabarcoding_miPEP_mutants/ITS/ITS/manip_mutants/export/mapping_file.tsv", sep="\t"), check.names=F)
newsample <- sample %>% mutate(Type = replace(Type, Type== "WT6.5", "WT"))
newsample <- newsample %>% mutate(Type = replace(Type, Type== "ago1", "ago1-27"))
newsample <- newsample %>% mutate(Type = replace(Type, Type== "dcl1", "dcl1-2"))
newsample <- newsample %>% mutate(Type = replace(Type, Type== "hen1", "hen1-4"))
newsample <- newsample %>% mutate(Type = replace(Type, Type== "Control", "Unplanted_soil"))
sample=sample_data(newsample)

rownames(sample)<-sample$`#SampleID`
merge<-merge_phyloseq(biom_file,sample)
merge
```

Removed WT14 samples: these samples are biologically unappropriate for our analysis, as they were not cultivated and sampled in the same conditions as the other samples. Therefore, only 2 wild type samples were appropriate for the following analyses, which was statistically detrimental.

```{r}
Samples_toRemove <- c(Type="WT14")
subset_samples(merge, Type %in% Samples_toRemove)
subset_samples(merge, !(Type%in%Samples_toRemove))

merge_noWT14 <- subset_samples(merge, !(Type%in%Samples_toRemove))
merge_noWT14 #should have only 48 samples (-2 WT14 roots, -2 WT14 soil)
```

Extracted and updated the taxa and OTU table.

```{r}
tax_table = tax_table(merge_noWT14)
#remplacer les noms de colonnes Rank1, rank2.. par les noms de taxons
colnames(tax_table) = c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species", "rank8", "rank9", "rank10", "rank11","rank12")
#write.table(tax_table, "tax_table.txt")

OTU_table = otu_table(merge_noWT14)
#write.table(OTU_table, "OTU_table.txt")

```

Counted the number of sequences in each sample. 

```{r}
Seq<-sample_sums(merge_noWT14)
Seq
#write.table(Seq, "sequences.txt")
min(Seq) # gives the lowest number of sequences in a sample = 132
max(Seq) # gives the highest number of sequences in a sample = 14 290
```

Kept ASVs that are at least present once in two samples (this step is needed, because we took out WT14 samples, some ASVs are only present in 1 or 2 samples). Two samples because, we have two WT roots and two WT soils (in case ASVs are specific to one or the other). Updated taxa and OTU table after.

```{r}
phylo_filtered = filter_taxa(merge_noWT14, function(x) sum(x >= 1)>= (2), TRUE)
OTU_table_phylo_filtered = otu_table(phylo_filtered)
#write.table(OTU_table_phylo_filtered, "OTU_table_phylo_filtered.txt")
tax_table_phylo_filtered = tax_table(phylo_filtered)
#write.table(tax_table_phylo_filtered, "tax_table_phylo_filtered.txt")
```

After this filtration, many ASVs (=taxa) are lost, but not so much sequences. This was checked by counting them again.

```{r}
Seq_filtered<-sample_sums(phylo_filtered)
Seq_filtered
#write.table(Seq_filtered, "sequences_filtered.txt")
```

Sequencing depth: do all samples reach a plateau? At which sample size?

```{r,fig.width = 10, fig.height = 7}
#remotes::install_github("mahendra-mariadassou/phyloseq-extended", ref = "dev")
library(phyloseq.extended)

p<-ggrare(phylo_filtered, step=100, color = "Compartment", label = "X.SampleID", se = FALSE)
p<-p + labs(title="Rarefaction curve", subtitle =  "ITS reads")
p<-p+scale_color_manual(values = c("chartreuse2", "dark orange"))
p<-p+theme_minimal()
p <- p + facet_wrap(~Type) 
p
```

ITS sequencing does not allow for "real" quantitative data. To overcome this, ITS amplicons were quantified by qPCR, along with a standard curve, to evaluate the "absolute" number of copies of ITS in each sample. The mean Ct was converted to a number of copies, by using the equation of the standard curve. The number of copies was used to replace the total sum of ITS ASVs, per sample, and each ASV was re-caculated as a number of copies. The resulting "absolute" abundance ASV table needed to be imported into R and processed into a new phyloseq object. 

1) Export phylo_filtered OTU table 
2) Open OTU table with Excel
    - align samples with their correct column
    - add "calculs" sheet and add "Samples" line and a "ASV ID" column
    - line 2 = number of ITS copies/sample (final number of copies, after multplied by ITS ratio=nb fungal ITS/total nb of ITS)
    - CALCULATION: to repeat for each sample: ASV = nb seq ASV * nb copies ITS / sum sequences in that sample 
3) Add a spread sheed to the Excel file "quant abs"
    -copy/paste the "calculs" sheet with just the data "1,2,3"
    -delete the line 2 (with the number of copies)
    -save as an Excel file (.xlsx)
4) Put all the cells into format "Number" with no decimals and export the active sheet as .txt (Tabulation)
5) Go to Galaxy.genouest.org and upload your .txt file, then search for "convert between BIOM table formats" input =.txt and output = BIOM1 file "OTU TABLE". 
6) Download the biom file and import to R 

```{r}
otu_biom <- import_biom("~/metabarcoding_miPEP_mutants/ITS/ITS/manip_mutants/otu_biom_ITS_mutants.biom1")

tax <- tax_table(tax_table_phylo_filtered) #taxa hasn't changed

samples_wo14 <- sample_data(merge_noWT14) #metadata hasn't changed

phylo_abs_ITS <- merge_phyloseq(tax,otu_biom,samples_wo14)

#Before continuing, we must discard the sample RTL1myc-3 from the root samples, because no valid Ct was measured in order to quantify the number of ITS copies in this sample.
Sample_toRemove <- c("X.SampleID"="RTL1myc.3-RACINE")
subset_samples(phylo_abs_ITS, X.SampleID %in% Sample_toRemove)
subset_samples(phylo_abs_ITS, !(X.SampleID %in% Sample_toRemove))
phylo_abs_ITS <- subset_samples(phylo_abs_ITS, !(X.SampleID %in% Sample_toRemove))
phylo_abs_ITS #should have only 47 samples now (-RTL1myc-3-RACINE)

tax_table = tax_table(phylo_abs_ITS)
#remplacer les noms de colonnes Rank1, rank2.. par les noms de taxons
colnames(tax_table) = c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species", "rank8", "rank9", "rank10", "rank11","rank12")
#write.table(tax_table, "tax_table.txt")

OTU_table = otu_table(phylo_abs_ITS)
samples_final <- sample_data(phylo_abs_ITS)

phylo_abs_ITS <- merge_phyloseq(tax_table,OTU_table,samples_final)

```

To always visualise the different mutants in the same order: a "correct order" was created and assigned to the phyloseq object. Root and soil samples were separated into two distinct phyloseq objects for the following analyses.

```{r}
correct.order <- c( "ago1-27", "dcl1-2", "hen1-4", "RTL1", "RTL1myc","WT","Unplanted_soil")
sample_data(phylo_abs_ITS)$Type <- factor(sample_data(phylo_abs_ITS)$Type, levels = correct.order)
levels(get_variable(phylo_abs_ITS, "Type")) #mettre dans le bon ordre

phylo_roots <- subset_samples(phylo_abs_ITS, Compartment=="RACINE")
phylo_soil <- subset_samples(phylo_abs_ITS, Compartment=="SOL")
```

ALPHA-DIVERSITY

What is the diversity within samples like? --> calculate alpha-diversity

The following is much inspired by https://www.nicholas-ollberding.com/post/introduction-to-the-statistical-analysis-of-microbiome-data-in-r/

The measurement of alpha-diversity is never 100% accurate, as (1) richness is always underestimated due to inexhaustive sampling and (2) rare taxa cannot be accounted for, as with NGS, singletons are eliminated from the beginning, as they could be just sequencing errors.
In order to compare the index between samples (they will have similar error rates) and so that sequencing depth doesn't interfer, it is recommended to rarefy the data.
Rarefaction is the subsampling of your reads without replacement to a same sequencing depth (=number of reads).

Before rarefying, plotting Total Reads VS Observed Richness can be interesting. It seems logical that the more reads in a sample, the more potential diversity can be observed.  

In the following paragraphs, the term "reads" is used, but it is really refering to "copies", as reads have been converted to copies, in the otu_biom.


```{r Rarefaction chunck part 2}
reads_soil <- sample_sums(phylo_soil)
min_reads_soil <- min(reads_soil)
min_reads_soil
#[1] 481 041 corresponds to hen1.4-D-SOL, the next lowest is at 1 325 266 RTL1myc-A-SOL: this will be the rarefaction limit, to not loose any more samples.

reads_roots <- sample_sums(phylo_roots)
min_reads_roots <- min(reads_roots)
min_reads_roots
#[1] 20 870 corresponds to WTdu6.5-2-RACINE, this sample cannot be discarded, so it will be the rarefaction limit. 

# Rarefying to a certain sample.size, is often reducing sequencing depth to the smallest sample. But if some samples are really small (low number of reads), it can be better to discard those samples and rarefy at a higher size. "rngseed" allows to set a seed, meaning that this code is reproducible (normally the random rarefaction will change every time the code is run!). "replace=FALSE" means that when a read is sampled it is not replaced in the pool of reads (https://www.thoughtco.com/sampling-with-or-without-replacement-3126563).

#sometimes the rarefaction process takes up too much memory: restart R and only run the essential scripts prior to rarefaction and then do it


testit(15)


```

Rarefaction

```{r}
# Roots samples 
phylo_rar_roots <- phyloseq::rarefy_even_depth(phylo_roots, sample.size = 20870, rngseed = 711, replace = FALSE)

testit(15)

# Soil samples
phylo_rar_soil <- phyloseq::rarefy_even_depth(phylo_soil, sample.size = 1325266 , rngseed = 711, replace = FALSE)

testit(10)

#lower samples are automatically removed from the analysis
```

Alpha-diversity in soil samples...

```{r, fig.width = 10, fig.height = 7}
p_Div_soil=plot_richness(phylo_rar_soil, color="Type", x="Type", measures=c("Observed","Shannon", "Simpson"))

p_Div_soil_bp=p_Div_soil+geom_boxplot(aes(color=Type), alpha=0.2)+ theme_classic()+xlab("Alpha-diversity in distant rhizosphere")+ylab("Observed ASVs richness")+ theme(axis.text.y = element_text (), axis.text.x = element_blank (), axis.text =element_text(size=8), axis.title=element_text(size=12,face="bold"))+scale_color_manual(values=c("#1B9E77", "#D95F02", "#7570B3", "#E7298A", "#66A61E", "#E6AB02", "#A6761D","#666666"))

p_Div_soil_bp

#Statistical analyses: be careful, if updating code: this part has to be manually (export/import)

alpha.diversity_soil=estimate_richness(phylo_rar_soil, measures=c("Observed","Shannon", "Simpson"))
alpha.diversity_soil
#write.table(alpha.diversity_soil, ("indice_diversite_soil.txt"))

#Add a "Type" column and re-import as Text(readr)

library(readr)
indice_diversite_soil <-read_delim("C:/Users/hmiddleton/Dropbox/Projet_Mutants_miPEPs_miRNA/Harriet/Metabarcoding_mutants_and_miPEP/Analyses_métabarcoding/mutants/ITS/indice_diversite_soil_1.txt","\t", escape_double = FALSE, trim_ws = TRUE)

indice_diversite_soil$Type <- as.factor(indice_diversite_soil$Type)
str(object = indice_diversite_soil$Type)
indice_diversite_soil$Type <- relevel(indice_diversite_soil$Type, ref="WT")

fit1s <-lm(Observed~Type, data=indice_diversite_soil)
hist(fit1s$residuals) #do residuals follow a normal distribution?
shapiro.test(fit1s$residuals)  # if p-value > 0,05 the distribution is not significantly different to a normal distribution. If p-value < 0,05 : a GLM test was used, as it does not presume normal distribution.
summary(fit1s)

fit2s <-lm(Shannon~Type, data=indice_diversite_soil)
hist(fit2s$residuals) 
shapiro.test(fit2s$residuals)
summary(fit2s)

fit3s <-lm(Simpson~Type, data=indice_diversite_soil)
hist(fit3s$residuals) 
shapiro.test(fit3s$residuals) 
glm_fit3s <- glm(Simpson~Type, data=indice_diversite_soil) #link="inverse" by default (Gamma "log" not ok because non-positive values)
summary(glm_fit3s)

```

Alpha-diversity in roots...

```{r, fig.width = 10, fig.height = 7}
p_Div_roots=plot_richness(phylo_rar_roots, color="Type", x="Type", measures=c("Observed", "Shannon", "Simpson"))

p_Div_roots_bp=p_Div_roots+geom_boxplot(aes(color=Type), alpha=0.2)+ theme_classic()+xlab("Alpha-diversity in roots and close rhizosphere")+ylab("Observed ASVs richness")+ theme(axis.text.y = element_text (), axis.text.x = element_blank (), axis.text =element_text(size=8), axis.title=element_text(size=12,face="bold"))+scale_color_manual(values=c("#1B9E77", "#D95F02", "#7570B3", "#E7298A", "#66A61E", "#E6AB02", "#A6761D","#666666"))

p_Div_roots_bp

#Statistical analyses: be careful, if updating code: this part has to be manually (export/import)

alpha.diversity_roots=estimate_richness(phylo_rar_roots, measures=c("Observed","Shannon", "Simpson"))

#write.table(alpha.diversity_roots, ("indice_diversite_roots.txt"))

#Add a "Type" column and re-import as Text(readr)

library(readr)
indice_diversite_roots <-read_delim("C:/Users/hmiddleton/Dropbox/Projet_Mutants_miPEPs_miRNA/Harriet/Metabarcoding_mutants_and_miPEP/Analyses_métabarcoding/mutants/ITS/indice_diversite_roots_1.txt","\t", escape_double = FALSE, trim_ws = TRUE)

indice_diversite_roots$Type <- as.factor(indice_diversite_roots$Type)
str(object = indice_diversite_roots$Type)
indice_diversite_roots$Type <- relevel(indice_diversite_roots$Type, ref="WT")

fit1 <-lm(Observed~Type, data=indice_diversite_roots)
hist(fit1$residuals) 
shapiro.test(fit1$residuals)
summary(fit1)

fit2 <-lm(Shannon~Type, data=indice_diversite_roots)
hist(fit2$residuals)
shapiro.test(fit2$residuals)
summary(fit2)

fit3 <-lm(Simpson~Type, data=indice_diversite_roots)
hist(fit3$residuals) 
shapiro.test(fit3$residuals)
glm_fit3 <- glm(Simpson~Type, data=indice_diversite_roots) #link="inverse" by default
summary(glm_fit3)

```

BETA-DIVERSITY

How different are these samples in comparison to each other?

1) Bray-Curtis distances: based on ASV abundance, not on phylogeny

Soil samples

```{r, fig.width = 10, fig.height = 7}
set.seed(711)
PCOA_Bray_soil <- ordinate(physeq = phylo_soil, method = "PCoA", distance="bray") 
PCOA_Bray_soil 
# The 2 main axes: Relative_eig have to be > Broken_stick, i.e. more than just random interpretation OR at least cumul eig > cumul broken stick
otu_soil <- otu_table(phylo_soil)
rda <- rda(otu_soil) #raw data without the sample column which conflicts w/rda
screeplot(rda, bstick=TRUE)
#This graph shows that the 2 first axes are > random

tiff("bc_soil_ITS.tiff", units="in", width=7, height=4, res=300)

bc_soil <- plot_ordination(physeq = phylo_soil, ordination=PCOA_Bray_soil, color="Type")+ theme_classic()+ theme(axis.text.y = element_text (), axis.text.x = element_blank (), axis.text =element_text(size=8), axis.title=element_text(size=12,face="bold"))+scale_color_manual(values=c("#1B9E77", "#D95F02", "#7570B3", "#E7298A", "#66A61E", "#E6AB02", "#A6761D","#666666"))+stat_ellipse()
bc_soil <- bc_soil + ggtitle("Bray-Curtis distances based PCoA of Arabidopsis thaliana mutants distant rhizospheric fungal communities") +theme(plot.title = element_text(size = 8, face = "bold"))
bc_soil
dev.off()

##################### Statistical analysis: permanova & beta-disper test
# Calculate bray curtis distance matrix
bray_soil <- phyloseq::distance(phylo_soil, method = "bray")
sampledf_soil <- data.frame(sample_data(phylo_soil))
adonis(bray_soil ~ Type, data = sampledf_soil)

# If Type has a significant effect, check group dispersion
Disper_Type_soil = betadisper(bray_soil, sampledf_soil$Type)
permutest(Disper_Type_soil, pairwise=TRUE, permutations=1000)

#If the dispersion test is signif, it means that the Type effect is actually only due to group dispersion, and not a real biological effect

#### Pairwise multilevel comparison using adonis #####
library(devtools)
#install_github("pmartinezarbizu/pairwiseAdonis/pairwiseAdonis") 

pairwise.adonis <- function(x,factors, sim.function = 'vegdist', sim.method = 'bray', p.adjust.m ='bonferroni',reduce=NULL,perm=999)
{
  
  co <- combn(unique(as.character(factors)),2)
  pairs <- c()
  Df <- c()
  SumsOfSqs <- c()
  F.Model <- c()
  R2 <- c()
  p.value <- c()
  
  
  for(elem in 1:ncol(co)){
    if(inherits(x, 'dist')){
      x1=as.matrix(x)[factors %in% c(as.character(co[1,elem]),as.character(co[2,elem])),
                      factors %in% c(as.character(co[1,elem]),as.character(co[2,elem]))]
    }
    
    else  (
      if (sim.function == 'daisy'){
        x1 = daisy(x[factors %in% c(co[1,elem],co[2,elem]),],metric=sim.method)
      } 
      else{x1 = vegdist(x[factors %in% c(co[1,elem],co[2,elem]),],method=sim.method)}
    )
    
    ad <- adonis(x1 ~ factors[factors %in% c(co[1,elem],co[2,elem])],
                 permutations = perm);
    pairs <- c(pairs,paste(co[1,elem],'vs',co[2,elem]));
    Df <- c(Df,ad$aov.tab[1,1])
    SumsOfSqs <- c(SumsOfSqs, ad$aov.tab[1,2])
    F.Model <- c(F.Model,ad$aov.tab[1,4]);
    R2 <- c(R2,ad$aov.tab[1,5]);
    p.value <- c(p.value,ad$aov.tab[1,6])
  }
  p.adjusted <- p.adjust(p.value,method=p.adjust.m)
  
  sig = c(rep('',length(p.adjusted)))
  sig[p.adjusted <= 0.05] <-'.'
  sig[p.adjusted <= 0.01] <-'*'
  sig[p.adjusted <= 0.001] <-'**'
  sig[p.adjusted <= 0.0001] <-'***'
  pairw.res <- data.frame(pairs,Df,SumsOfSqs,F.Model,R2,p.value,p.adjusted,sig)
  
  if(!is.null(reduce)){
    pairw.res <- subset (pairw.res, grepl(reduce,pairs))
    pairw.res$p.adjusted <- p.adjust(pairw.res$p.value,method=p.adjust.m)
    
    sig = c(rep('',length(pairw.res$p.adjusted)))
    sig[pairw.res$p.adjusted <= 0.1] <-'.'
    sig[pairw.res$p.adjusted <= 0.05] <-'*'
    sig[pairw.res$p.adjusted <= 0.01] <-'**'
    sig[pairw.res$p.adjusted <= 0.001] <-'***'
    pairw.res <- data.frame(pairw.res[,1:7],sig)
  }
  class(pairw.res) <- c("pwadonis", "data.frame")
  return(pairw.res)
} 


### Method summary
summary.pwadonis = function(object, ...) {
  cat("Result of pairwise.adonis:\n")
  cat("\n")
  print(object, ...)
  cat("\n")
  cat("Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1\n")
}

# All Types
pairwise.adonis(bray_soil, sampledf_soil$Type)

```

Root samples

```{r, fig.width = 10, fig.height = 7}
set.seed(711)

PCOA_Bray_roots <- ordinate(physeq = phylo_roots, method = "PCoA", distance="bray") 
PCOA_Bray_roots 
# The 2 main axes: Relative_eig have to be > Broken_stick, i.e. more than just random interpretation OR at least cumul eig > cumul broken stick
otu_roots <- otu_table(phylo_roots)
rda <- rda(otu_roots) #raw data without the sample column which conflicts w/rda
screeplot(rda, bstick=TRUE)
#This graph shows that the 2nd axis is not interpretable

tiff("bc_roots_ITS.tiff", units="in", width=7, height=4, res=300)

bc_roots <- plot_ordination(physeq = phylo_roots, ordination=PCOA_Bray_roots, color="Type")+ theme_classic()+ theme(axis.text.y = element_text (), axis.text.x = element_blank (), axis.text =element_text(size=8), axis.title=element_text(size=12,face="bold"))+scale_color_manual(values=c("#1B9E77", "#D95F02", "#7570B3", "#E7298A", "#66A61E", "#E6AB02", "#A6761D","#666666"))+stat_ellipse()
bc_roots <- bc_roots + ggtitle("Bray-Curtis distances based PCoA of Arabidopsis thaliana mutants of roots and close rhizospheric fungal communities") +theme(plot.title = element_text(size = 8, face = "bold"))
bc_roots
dev.off()

#################### Statistical analysis: permanova & beta-disper test
# Calculate bray curtis distance matrix
bray_roots <- phyloseq::distance(phylo_roots, method = "bray")
sampledf_roots <- data.frame(sample_data(phylo_roots))
adonis(bray_roots ~ Type, data = sampledf_roots)

# If Type has a significant effect, check group dispersion
Disper_Type_roots = betadisper(bray_roots, sampledf_roots$Type)
permutest(Disper_Type_roots, pairwise=TRUE, permutations=1000)

#If the dispersion test is signif, it means that the Type effect is actually only due to group dispersion, and not a real biological effect

pairwise.adonis(bray_roots, sampledf_roots$Type)
```

2) Weighted UniFrac distances: based on ASV abundance and phylogeny

  a) First, the provided phylogenetic tree was added to the phyloseq object
 
```{r}
library("ape")
library(magrittr)

pick_new_outgroup <- function(tree.unrooted){
  require("magrittr")
  require("data.table")
  require("ape") # ape::Ntip
  # tablify parts of tree that we need.
  treeDT <-
    cbind(
      data.table(tree.unrooted$edge),
      data.table(length = tree.unrooted$edge.length)
    )[1:Ntip(tree.unrooted)] %>%
    cbind(data.table(id = tree.unrooted$tip.label))
  # Take the longest terminal branch as outgroup
  new.outgroup <- treeDT[which.max(length)]$id
  return(new.outgroup) }
  
tree <- read.tree("~/metabarcoding_miPEP_mutants/ITS/ITS/manip_mutants/export/tree/tree.fasttree")
tree
my.tree <- phy_tree(tree)
out.group <- pick_new_outgroup(my.tree)
out.group #ex : [1] "3312"
new.tree <- ape::root(my.tree, outgroup=out.group, resolve.root=TRUE)#root tree to always have the same PCoA at each re-run
new.tree
#use the latest phyloseq info (-WT14-RTL1myc-racine)
phylo_for_UniFrac <- phyloseq(tax_table,OTU_table,samples_final, new.tree)
phylo_for_UniFrac
```

  b)This phyloseq was subsetted into phylo_for_UniFrac_roots and "_soil
  
```{r, fig.width = 10, fig.height = 7}
correct.order <- c( "ago1-27", "dcl1-2", "hen1-4", "RTL1", "RTL1myc","WT","Unplanted_soil")
sample_data(phylo_for_UniFrac)$Type <- factor(sample_data(phylo_for_UniFrac)$Type, levels = correct.order)
levels(get_variable(phylo_for_UniFrac, "Type")) 

phylo_for_UniFrac_soil <- subset_samples(phylo_for_UniFrac, Compartment=="SOL")
phylo_for_UniFrac_roots <- subset_samples(phylo_for_UniFrac, Compartment=="RACINE")
```

 c) WUF distances were calculated and represented as Principal Coordinate Analysis (PCoA)
 
 Soil samples
 
```{r, fig.width = 10, fig.height = 7}
set.seed(711)
PCOA_weighted_soil <- ordinate(physeq = phylo_for_UniFrac_soil,  method = "PCoA", distance="unifrac", weighted=TRUE) 
PCOA_weighted_soil 
# The 2 main axes: Relative_eig have to be > Broken_stick, i.e. more than just random interpretation OR at least cumul eig > cumul broken stick
# the second axis is not interpretable
otu_w_soil <- otu_table(phylo_for_UniFrac_soil)
rda <- rda(otu_w_soil) #raw data without the sample column which conflicts w/rda
screeplot(rda, bstick=TRUE)
#This graph shows that the 2 first axes are > random

tiff("WUF_soil_ITS.tiff", units="in", width=7, height=4, res=300)


WUF_soil <- plot_ordination(physeq = phylo_for_UniFrac_soil, ordination=PCOA_weighted_soil, color="Type")+ theme_classic()+ theme(axis.text.y = element_text (), axis.text.x = element_blank (), axis.text =element_text(size=8), axis.title=element_text(size=12,face="bold"))+scale_color_manual(values=c("#1B9E77", "#D95F02", "#7570B3", "#E7298A", "#66A61E", "#E6AB02", "#A6761D","#666666"))+stat_ellipse()
WUF_soil <- WUF_soil + ggtitle("Weighted UniFrac distances based PCoA of Arabidopsis thaliana mutants distant rhizospheric fungal communities") +theme(plot.title = element_text(size = 8, face = "bold"))
WUF_soil
dev.off()

################### Statistical analysis: permanova & beta-disper test

# Calculate distance matrix
WUF_soil <- phyloseq::distance(phylo_for_UniFrac_soil, method = "unifrac", weighted=TRUE)
sampledf_soil <- data.frame(sample_data(phylo_for_UniFrac_soil))
adonis(WUF_soil ~ Type, data = sampledf_soil)

# If Type has a significant effect, check group dispersion
Disper_Type_soil = betadisper(WUF_soil, sampledf_soil$Type)
permutest(Disper_Type_soil, pairwise=TRUE, permutations=1000)

#If the dispersion test is signif, it means that the Type effect is actually only due to group dispersion, and not a real biological effect

pairwise.adonis(WUF_soil, sampledf_soil$Type)
```

Root samples

```{r, fig.width = 10, fig.height = 7}
set.seed(711)
PCOA_weighted_roots <- ordinate(physeq = phylo_for_UniFrac_roots,  method = "PCoA", distance="unifrac", weighted=TRUE) 
PCOA_weighted_roots 
# The 2 main axes: Relative_eig has to be > Broken_stick, i.e. more than just random interpretation OR at least cumul eig > cumul broken stick
otu_w_roots <- otu_table(phylo_for_UniFrac_roots)
rda <- rda(otu_w_roots) #raw data without the sample column which conflicts w/rda
screeplot(rda, bstick=TRUE)
#This graph shows that the 2nd axis is random

tiff("WUF_roots_ITS.tiff", units="in", width=7, height=4, res=300)


WUF_roots <- plot_ordination(physeq = phylo_for_UniFrac_roots, ordination=PCOA_weighted_roots, color="Type")+ theme_classic()+ theme(axis.text.y = element_text (), axis.text.x = element_blank (), axis.text =element_text(size=8), axis.title=element_text(size=12,face="bold"))+scale_color_manual(values=c("#1B9E77", "#D95F02", "#7570B3", "#E7298A", "#66A61E", "#E6AB02", "#A6761D","#666666"))+stat_ellipse()
WUF_roots <- WUF_roots + ggtitle("Weighted UniFrac distances based PCoA of Arabidopsis thaliana mutants of roots and close rhizospheric fungal communities") +theme(plot.title = element_text(size = 8, face = "bold"))
WUF_roots
dev.off()

################## Statistical analysis: permanova & beta-disper test

# Calculate distance matrix
WUF_roots <- phyloseq::distance(phylo_for_UniFrac_roots, method = "unifrac", weighted=TRUE)
sampledf_roots <- data.frame(sample_data(phylo_for_UniFrac_roots))
adonis(WUF_roots ~ Type, data = sampledf_roots)

# If Type has a significant effect, check group dispersion
Disper_Type_roots = betadisper(WUF_roots, sampledf_roots$Type)
permutest(Disper_Type_roots, pairwise=TRUE, permutations=1000)

#If the dispersion test is signif, it means that the Type effect is actually only due to group dispersion, and not a real biological effect

pairwise.adonis(WUF_roots, sampledf_roots$Type)
```

DIFFERENTIALLY ABUNDANT ASVs: ENRICHMENT AND DEPLETION COMPARED WITH WT

Another type of analysis is to look at OTUs/ASVs that are specifically enriched or depleted in certain sample Types 

1) Soil samples

```{r, fig.width = 10, fig.height = 7}
library(DESeq2) # The DESeq2 package uses the raw OTU/ASV table (non rarefied phyloseq) and calculates a negative binomial GLM for each OTU/ASV. The Wald test was used for statistics.

#Convert phyloseq object into a DESeqDataSet object with dispersion estimates using "Type" as the main factor
ds2_soil <- phyloseq_to_deseq2(phylo_soil, ~ Type) 
#Specify your reference group, or else it will use alphabetical order
ds2_soil$Type <- relevel(ds2_soil$Type, ref = "WT")

#calculate geometric means prior to estimate size factors
gm_mean = function(x, na.rm=TRUE){exp(sum(log(x[x > 0]), na.rm=na.rm) / length(x))}
geoMeans = apply(counts(ds2_soil), 1, gm_mean)
ds2_soil= estimateSizeFactors(ds2_soil, geoMeans = geoMeans)

dds_soil <- DESeq(ds2_soil, test="Wald", fitType="parametric")

```

For each Type x, ASVs which are differentially abundant in samples of Type x compared with wild type samples (=reference) are searched and statistically tested.

ago1-27 in soil samples

```{r, fig.width = 10, fig.height = 7}
# Note that cooksCutoff = FALSE, because it calculates outliers only when 2+ replicates. As WT only has 2 replicates and that outliers cannot be calculated, they are not calculated for others either
res_soil_ago <- results(dds_soil, cooksCutoff = FALSE, contrast = c("Type","WT","ago1-27")) # gives the table showing the comparison of log2FoldChange between the "treated" and "untreated" group
res_soil_ago_order = res_soil_ago[order(res_soil_ago$padj, na.last=NA), ] # order results by adjusted p-value (Benjamini-Hochberg correction)

alpha = 0.001
sigtab_soil_ago = res_soil_ago_order[which(res_soil_ago_order$padj < alpha),]
sigtab_soil_ago = cbind(as(sigtab_soil_ago, "data.frame"), as(tax_table(phylo_soil)[rownames(sigtab_soil_ago), ], "matrix"))
dim(sigtab_soil_ago)
write.table(sigtab_soil_ago, "ago_soil.txt")
########### Graphic Representation ######################
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
    scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab_soil_ago$log2FoldChange, sigtab_soil_ago$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_soil_ago$Phylum = factor(as.character(sigtab_soil_ago$Phylum), levels=names(x))
# Genus order
x = tapply(sigtab_soil_ago$log2FoldChange, sigtab_soil_ago$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_soil_ago$Genus = factor(as.character(sigtab_soil_ago$Genus), levels=names(x))
plot_ago_soil <- ggplot(sigtab_soil_ago, aes(x=Genus, y=log2FoldChange, color=Phylum)) + geom_point(size=1) +theme(axis.text.x = element_text(size=8, angle = -90, hjust = 0, vjust=0.5), legend.text = element_text(size = 10))+ geom_hline(yintercept=0, color="black") + labs(title = "Differentially abundant ASVs in comparison with WT (p-adj<0.001)",subtitle = "Mutants ago1-27 distant rhizosphere")
plot_ago_soil

```

This step was repeated for each Type.

dcl1-2 in soil samples
```{r, fig.width = 10, fig.height = 7}
res_soil_dcl <- results(dds_soil, cooksCutoff = FALSE, contrast = c("Type","WT","dcl1-2"))
res_soil_dcl_order = res_soil_dcl[order(res_soil_dcl$padj, na.last=NA), ]
alpha = 0.001
sigtab_soil_dcl = res_soil_dcl_order[which(res_soil_dcl_order$padj < alpha),]
sigtab_soil_dcl = cbind(as(sigtab_soil_dcl, "data.frame"), as(tax_table(phylo_soil)[rownames(sigtab_soil_dcl), ], "matrix"))
dim(sigtab_soil_dcl)
write.table(sigtab_soil_dcl, "dcl_soil.txt")
########### Graphic Representation ######################
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
    scale_fill_brewer(palette = palname, ...)}
x = tapply(sigtab_soil_dcl$log2FoldChange, sigtab_soil_dcl$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_soil_dcl$Phylum = factor(as.character(sigtab_soil_dcl$Phylum), levels=names(x))
x = tapply(sigtab_soil_dcl$log2FoldChange, sigtab_soil_dcl$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_soil_dcl$Genus = factor(as.character(sigtab_soil_dcl$Genus), levels=names(x))
plot_dcl_soil <- ggplot(sigtab_soil_dcl, aes(x=Genus, y=log2FoldChange, color=Phylum)) + geom_point(size=1) +theme(axis.text.x = element_text(size=8, angle = -90, hjust = 0, vjust=0.5), legend.text = element_text(size = 10))+ geom_hline(yintercept=0, color="black") + labs(title = " differentially abundant ASVs in comparison with WT (p-adj<0.001)",subtitle = "Mutants dcl1-2 distant rhizosphere")
plot_dcl_soil

```

hen1-4 soil samples
```{r, fig.width = 10, fig.height = 7}
res_soil_hen <- results(dds_soil, cooksCutoff = FALSE, contrast = c("Type","WT","hen1-4"))
res_soil_hen_order = res_soil_hen[order(res_soil_hen$padj, na.last=NA), ]
alpha = 0.001
sigtab_soil_hen = res_soil_hen_order[which(res_soil_hen_order$padj < alpha),]
sigtab_soil_hen = cbind(as(sigtab_soil_hen, "data.frame"), as(tax_table(phylo_soil)[rownames(sigtab_soil_hen), ], "matrix"))
dim(sigtab_soil_hen)
write.table(sigtab_soil_hen, "hen_soil.txt")
########### Graphic Representation ######################
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
    scale_fill_brewer(palette = palname, ...)}
x = tapply(sigtab_soil_hen$log2FoldChange, sigtab_soil_hen$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_soil_hen$Phylum = factor(as.character(sigtab_soil_hen$Phylum), levels=names(x))
x = tapply(sigtab_soil_hen$log2FoldChange, sigtab_soil_hen$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_soil_hen$Genus = factor(as.character(sigtab_soil_hen$Genus), levels=names(x))
plot_hen_soil <- ggplot(sigtab_soil_hen, aes(x=Genus, y=log2FoldChange, color=Phylum)) + geom_point(size=1) +theme(axis.text.x = element_text(size=8, angle = -90, hjust = 0, vjust=0.5), legend.text = element_text(size = 10))+ geom_hline(yintercept=0, color="black") + labs(title = " differentially abundant ASVs in comparison with WT (p-adj<0.001)",subtitle = "Mutants hen1-4 distant rhizosphere")
plot_hen_soil

```

RTL1 soil samples
```{r, fig.width = 10, fig.height = 7}
res_soil_RTL1 <- results(dds_soil, cooksCutoff = FALSE, contrast = c("Type","WT","RTL1"))
res_soil_RTL1_order = res_soil_RTL1[order(res_soil_RTL1$padj, na.last=NA), ]
alpha = 0.001
sigtab_soil_RTL1 = res_soil_RTL1_order[which(res_soil_RTL1_order$padj < alpha),]
sigtab_soil_RTL1 = cbind(as(sigtab_soil_RTL1, "data.frame"), as(tax_table(phylo_soil)[rownames(sigtab_soil_RTL1), ], "matrix"))
dim(sigtab_soil_RTL1)
write.table(sigtab_soil_RTL1, "RTL1_soil.txt")
########### Graphic Representation ######################
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
    scale_fill_brewer(palette = palname, ...)}
x = tapply(sigtab_soil_RTL1$log2FoldChange, sigtab_soil_RTL1$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_soil_RTL1$Phylum = factor(as.character(sigtab_soil_RTL1$Phylum), levels=names(x))
x = tapply(sigtab_soil_RTL1$log2FoldChange, sigtab_soil_RTL1$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_soil_RTL1$Genus = factor(as.character(sigtab_soil_RTL1$Genus), levels=names(x))
plot_RTL1_soil <- ggplot(sigtab_soil_RTL1, aes(x=Genus, y=log2FoldChange, color=Phylum)) + geom_point(size=1) +theme(axis.text.x = element_text(size=8, angle = -90, hjust = 0, vjust=0.5), legend.text = element_text(size = 10))+ geom_hline(yintercept=0, color="black") + labs(title = " differentially abundant ASVs in comparison with WT (p-adj<0.001)",subtitle = "Mutants RTL1 distant rhizosphere")
plot_RTL1_soil
```

RTL1myc soil samples
```{r, fig.width = 10, fig.height = 7}
res_soil_RTL1myc <- results(dds_soil, cooksCutoff = FALSE, contrast = c("Type","WT","RTL1myc"))
res_soil_RTL1myc_order = res_soil_RTL1myc[order(res_soil_RTL1myc$padj, na.last=NA), ]
alpha = 0.001
sigtab_soil_RTL1myc = res_soil_RTL1myc_order[which(res_soil_RTL1myc_order$padj < alpha),]
sigtab_soil_RTL1myc = cbind(as(sigtab_soil_RTL1myc, "data.frame"), as(tax_table(phylo_soil)[rownames(sigtab_soil_RTL1myc), ], "matrix"))
dim(sigtab_soil_RTL1myc)
write.table(sigtab_soil_RTL1myc, "RTL1myc_soil.txt")
########### Graphic Representation ######################
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
    scale_fill_brewer(palette = palname, ...)}
x = tapply(sigtab_soil_RTL1myc$log2FoldChange, sigtab_soil_RTL1myc$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_soil_RTL1myc$Phylum = factor(as.character(sigtab_soil_RTL1myc$Phylum), levels=names(x))
x = tapply(sigtab_soil_RTL1myc$log2FoldChange, sigtab_soil_RTL1myc$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_soil_RTL1myc$Genus = factor(as.character(sigtab_soil_RTL1myc$Genus), levels=names(x))
plot_RTL1myc_soil <- ggplot(sigtab_soil_RTL1myc, aes(x=Genus, y=log2FoldChange, color=Phylum)) + geom_point(size=1) +theme(axis.text.x = element_text(size=8, angle = -90, hjust = 0, vjust=0.5), legend.text = element_text(size = 10))+ geom_hline(yintercept=0, color="black") + labs(title = " differentially abundant ASVs in comparison with WT (p-adj<0.001)",subtitle = "Mutants RTL1myc distant rhizosphere")
plot_RTL1myc_soil
```

Unplanted_soil samples
```{r, fig.width = 10, fig.height = 7}
res_soil_Unplanted_soil <- results(dds_soil, cooksCutoff = FALSE, contrast = c("Type","WT","Unplanted_soil"))
res_soil_Unplanted_soil_order = res_soil_Unplanted_soil[order(res_soil_Unplanted_soil$padj, na.last=NA), ]
alpha = 0.001
sigtab_soil_Unplanted_soil = res_soil_Unplanted_soil_order[which(res_soil_Unplanted_soil_order$padj < alpha),]
sigtab_soil_Unplanted_soil = cbind(as(sigtab_soil_Unplanted_soil, "data.frame"), as(tax_table(phylo_soil)[rownames(sigtab_soil_Unplanted_soil), ], "matrix"))
dim(sigtab_soil_Unplanted_soil)
write.table(sigtab_soil_Unplanted_soil, "unplanted_soil.txt")
########### Graphic Representation ######################
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
    scale_fill_brewer(palette = palname, ...)}
x = tapply(sigtab_soil_Unplanted_soil$log2FoldChange, sigtab_soil_Unplanted_soil$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_soil_Unplanted_soil$Phylum = factor(as.character(sigtab_soil_Unplanted_soil$Phylum), levels=names(x))
x = tapply(sigtab_soil_Unplanted_soil$log2FoldChange, sigtab_soil_Unplanted_soil$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_soil_Unplanted_soil$Genus = factor(as.character(sigtab_soil_Unplanted_soil$Genus), levels=names(x))
plot_Unplanted_soil_soil <- ggplot(sigtab_soil_Unplanted_soil, aes(x=Genus, y=log2FoldChange, color=Phylum)) + geom_point(size=1) +theme(axis.text.x = element_text(size=8, angle = -90, hjust = 0, vjust=0.5), legend.text = element_text(size = 10))+ geom_hline(yintercept=0, color="black") + labs(title = " differentially abundant ASVs in comparison with WT (p-adj<0.001)",subtitle = "Mutants Unplanted_soil distant rhizosphere")
plot_Unplanted_soil_soil
```

2) Root samples

```{r}
library(DESeq2)
ds2_roots <- phyloseq_to_deseq2(phylo_roots, ~ Type) 
ds2_roots$Type <- relevel(ds2_roots$Type, ref = "WT")
gm_mean = function(x, na.rm=TRUE){exp(sum(log(x[x > 0]), na.rm=na.rm) / length(x))}
geoMeans = apply(counts(ds2_roots), 1, gm_mean)
ds2_roots= estimateSizeFactors(ds2_roots, geoMeans = geoMeans)
dds_roots <- DESeq(ds2_roots, test="Wald", fitType="parametric")

```

ago1-27 in roots

```{r, fig.width = 10, fig.height = 7}
res_roots_ago <- results(dds_roots, cooksCutoff = FALSE, contrast = c("Type","WT","ago1-27")) 
res_roots_ago_order = res_roots_ago[order(res_roots_ago$padj, na.last=NA), ] 

alpha = 0.001
sigtab_roots_ago = res_roots_ago_order[which(res_roots_ago_order$padj < alpha),]
sigtab_roots_ago = cbind(as(sigtab_roots_ago, "data.frame"), as(tax_table(phylo_roots)[rownames(sigtab_roots_ago), ], "matrix"))
dim(sigtab_roots_ago)
write.table(sigtab_roots_ago, "ago_roots.txt")
########### Graphic Representation ######################
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
    scale_fill_brewer(palette = palname, ...)}
x = tapply(sigtab_roots_ago$log2FoldChange, sigtab_roots_ago$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_roots_ago$Phylum = factor(as.character(sigtab_roots_ago$Phylum), levels=names(x))
x = tapply(sigtab_roots_ago$log2FoldChange, sigtab_roots_ago$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_roots_ago$Genus = factor(as.character(sigtab_roots_ago$Genus), levels=names(x))
plot_ago_roots <- ggplot(sigtab_roots_ago, aes(x=Genus, y=log2FoldChange, color=Phylum)) + geom_point(size=1) +theme(axis.text.x = element_text(size=8, angle = -90, hjust = 0, vjust=0.5), legend.text = element_text(size = 10))+ geom_hline(yintercept=0, color="black") + labs(title = " differentially abundant ASVs in comparison with WT (p-adj<0.001)",subtitle = "Mutants ago1-27 in roots and close rhizosphere")
plot_ago_roots
```

dcl1-2 in roots

```{r, fig.width = 10, fig.height = 7}
res_roots_dcl <- results(dds_roots, cooksCutoff = FALSE, contrast = c("Type","WT","dcl1-2")) 
res_roots_dcl_order = res_roots_dcl[order(res_roots_dcl$padj, na.last=NA), ] 

alpha = 0.001
sigtab_roots_dcl = res_roots_dcl_order[which(res_roots_dcl_order$padj < alpha),]
sigtab_roots_dcl = cbind(as(sigtab_roots_dcl, "data.frame"), as(tax_table(phylo_roots)[rownames(sigtab_roots_dcl), ], "matrix"))
dim(sigtab_roots_dcl)
write.table(sigtab_roots_dcl, "dcl_roots.txt")
########### Graphic Representation ######################
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
    scale_fill_brewer(palette = palname, ...)}
x = tapply(sigtab_roots_dcl$log2FoldChange, sigtab_roots_dcl$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_roots_dcl$Phylum = factor(as.character(sigtab_roots_dcl$Phylum), levels=names(x))
x = tapply(sigtab_roots_dcl$log2FoldChange, sigtab_roots_dcl$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_roots_dcl$Genus = factor(as.character(sigtab_roots_dcl$Genus), levels=names(x))
plot_dcl_roots <- ggplot(sigtab_roots_dcl, aes(x=Genus, y=log2FoldChange, color=Phylum)) + geom_point(size=1) +theme(axis.text.x = element_text(size=8, angle = -90, hjust = 0, vjust=0.5), legend.text = element_text(size = 10))+ geom_hline(yintercept=0, color="black") + labs(title = " differentially abundant ASVs in comparison with WT (p-adj<0.001)",subtitle = "Mutants dcl1-2 in roots and close rhizosphere")
plot_dcl_roots
```

hen1-4 in roots

```{r, fig.width = 10, fig.height = 7}
res_roots_hen <- results(dds_roots, cooksCutoff = FALSE, contrast = c("Type","WT","hen1-4")) 
res_roots_hen_order = res_roots_hen[order(res_roots_hen$padj, na.last=NA), ] 

alpha = 0.001
sigtab_roots_hen = res_roots_hen_order[which(res_roots_hen_order$padj < alpha),]
sigtab_roots_hen = cbind(as(sigtab_roots_hen, "data.frame"), as(tax_table(phylo_roots)[rownames(sigtab_roots_hen), ], "matrix"))
dim(sigtab_roots_hen)
write.table(sigtab_roots_hen, "hen_roots.txt")
########### Graphic Representation ######################
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
    scale_fill_brewer(palette = palname, ...)}
x = tapply(sigtab_roots_hen$log2FoldChange, sigtab_roots_hen$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_roots_hen$Phylum = factor(as.character(sigtab_roots_hen$Phylum), levels=names(x))
x = tapply(sigtab_roots_hen$log2FoldChange, sigtab_roots_hen$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_roots_hen$Genus = factor(as.character(sigtab_roots_hen$Genus), levels=names(x))
plot_hen_roots <- ggplot(sigtab_roots_hen, aes(x=Genus, y=log2FoldChange, color=Phylum)) + geom_point(size=1) +theme(axis.text.x = element_text(size=8, angle = -90, hjust = 0, vjust=0.5), legend.text = element_text(size = 10))+ geom_hline(yintercept=0, color="black") + labs(title = " differentially abundant ASVs in comparison with WT (p-adj<0.001)",subtitle = "Mutants hen1-4 in roots and close rhizosphere")
plot_hen_roots
```

RTL1 in roots

```{r, fig.width = 10, fig.height = 7}
res_roots_RTL1 <- results(dds_roots, cooksCutoff = FALSE, contrast = c("Type","WT","RTL1")) 
res_roots_RTL1_order = res_roots_RTL1[order(res_roots_RTL1$padj, na.last=NA), ] 

alpha = 0.001
sigtab_roots_RTL1 = res_roots_RTL1_order[which(res_roots_RTL1_order$padj < alpha),]
sigtab_roots_RTL1 = cbind(as(sigtab_roots_RTL1, "data.frame"), as(tax_table(phylo_roots)[rownames(sigtab_roots_RTL1), ], "matrix"))
dim(sigtab_roots_RTL1)
write.table(sigtab_roots_RTL1, "RTL1_roots.txt")
########### Graphic Representation ######################
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
    scale_fill_brewer(palette = palname, ...)}
x = tapply(sigtab_roots_RTL1$log2FoldChange, sigtab_roots_RTL1$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_roots_RTL1$Phylum = factor(as.character(sigtab_roots_RTL1$Phylum), levels=names(x))
x = tapply(sigtab_roots_RTL1$log2FoldChange, sigtab_roots_RTL1$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_roots_RTL1$Genus = factor(as.character(sigtab_roots_RTL1$Genus), levels=names(x))
plot_RTL1_roots <- ggplot(sigtab_roots_RTL1, aes(x=Genus, y=log2FoldChange, color=Phylum)) + geom_point(size=1) +theme(axis.text.x = element_text(size=8, angle = -90, hjust = 0, vjust=0.5), legend.text = element_text(size = 10))+ geom_hline(yintercept=0, color="black") + labs(title = " differentially abundant ASVs in comparison with WT (p-adj<0.001)",subtitle = "Mutants RTL1 in roots and close rhizosphere")
plot_RTL1_roots
```

RTL1myc in roots

```{r, fig.width = 10, fig.height = 7}
res_roots_RTL1myc <- results(dds_roots, cooksCutoff = FALSE, contrast = c("Type","WT","RTL1myc")) 
res_roots_RTL1myc_order = res_roots_RTL1myc[order(res_roots_RTL1myc$padj, na.last=NA), ] 

alpha = 0.001
sigtab_roots_RTL1myc = res_roots_RTL1myc_order[which(res_roots_RTL1myc_order$padj < alpha),]
sigtab_roots_RTL1myc = cbind(as(sigtab_roots_RTL1myc, "data.frame"), as(tax_table(phylo_roots)[rownames(sigtab_roots_RTL1myc), ], "matrix"))
dim(sigtab_roots_RTL1myc)
write.table(sigtab_roots_RTL1myc, "RTL1myc_roots.txt")
########### Graphic Representation ######################
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
    scale_fill_brewer(palette = palname, ...)}
x = tapply(sigtab_roots_RTL1myc$log2FoldChange, sigtab_roots_RTL1myc$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_roots_RTL1myc$Phylum = factor(as.character(sigtab_roots_RTL1myc$Phylum), levels=names(x))
x = tapply(sigtab_roots_RTL1myc$log2FoldChange, sigtab_roots_RTL1myc$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_roots_RTL1myc$Genus = factor(as.character(sigtab_roots_RTL1myc$Genus), levels=names(x))
plot_RTL1myc_roots <- ggplot(sigtab_roots_RTL1myc, aes(x=Genus, y=log2FoldChange, color=Phylum)) + geom_point(size=1) +theme(axis.text.x = element_text(size=8, angle = -90, hjust = 0, vjust=0.5), legend.text = element_text(size = 10))+ geom_hline(yintercept=0, color="black") + labs(title = " differentially abundant ASVs in comparison with WT (p-adj<0.001)",subtitle = "Mutants RTL1myc in roots and close rhizosphere")
plot_RTL1myc_roots
```

INDICATOR SPECIES

Are some species indicative of certain mutants ? 

1) Soil Samples

```{r, fig.width = 10, fig.height = 7}
# https://jkzorz.github.io/2019/07/02/Indicator-species-analysis.html
#library(indicspecies)

# Export the OTU table, with a Sample column, Type column and then ASV & abundances per column
#OTU_table_phylo_soil <- otu_table(phylo_soil)
#head(OTU_table_phylo_soil)
#write.table(OTU_table_phylo_soil, "OTU_table_phylo_soil.txt")

#Re-import the table OTU_table_phylo_soil_1.xlsx
#OTU_table_phylo_soil_1<-read_excel("C:/Users/hmiddleton/Dropbox/Projet_Mutants_miPEPs_miRNA/Harriet/Metabarcoding_mutants_and_miPEP/Analyses_métabarcoding/mutants/ITS/OTU_table_phylo_soil_1.xlsx")

#abund_soil = OTU_table_phylo_soil_1[,3:ncol(OTU_table_phylo_soil_1)] #ASV abundance starts at column 3
#Type = OTU_table_phylo_soil_1$Type # create a vector that contains the information from the "Type" column
#set.seed(771)
#inv_soil = multipatt(abund_soil, Type, func = "r.g", control = how(nperm=9999))
#summary(inv_soil)

#IndVal.g is the corrected version of IndVal for unequal group sizes. "r." measures correlation between two binary vectors: r.g. is a correction of this measurement which uses abundance and not just presence/absence	

#A = specificity or positive predictive value (probability that the investigated group does belong to the target group, due to the presence of indic species) 

# B= fidelity or sensitivity (probability that indic species belongs to target group) 
```

2) Root samples

```{r, fig.width = 10, fig.height = 7}
#library(indicspecies)

# Export the OTU table, with a Sample column, Type column and then ASV & abundances per column
#OTU_table_phylo_roots <- otu_table(phylo_roots)
#head(OTU_table_phylo_roots)
#write.table(OTU_table_phylo_roots, "OTU_table_phylo_roots.txt")

#Re-import the table OTU_table_phylo_roots_1.xlsx
#OTU_table_phylo_roots_1<-read_excel("C:/Users/hmiddleton/Dropbox/Projet_Mutants_miPEPs_miRNA/Harriet/Metabarcoding_mutants_and_miPEP/Analyses_métabarcoding/mutants/ITS/OTU_table_phylo_roots_1.xlsx")

#abund_roots = OTU_table_phylo_roots_1[,3:ncol(OTU_table_phylo_roots_1)] #ASV abundance starts at column 3
#Type = OTU_table_phylo_roots_1$Type # create a vector that contains the information from the "Type" column
#set.seed(771)
#inv_roots = multipatt(abund_roots, Type, func = "r.g", control = how(nperm=9999))
#summary(inv_roots)

```

